<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HashSearchClass</class>
 <widget class="QMainWindow" name="HashSearchClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>785</width>
    <height>656</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>651</width>
    <height>539</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>HashSearch</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>290</y>
      <width>761</width>
      <height>101</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>姓名</string>
     </property>
     <property name="font">
      <font>
       <family>宋体</family>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>电话号码</string>
     </property>
     <property name="font">
      <font>
       <family>宋体</family>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>邮箱</string>
     </property>
     <property name="font">
      <font>
       <family>宋体</family>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>地址</string>
     </property>
     <property name="font">
      <font>
       <family>宋体</family>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>查询耗时(ms)</string>
     </property>
     <property name="font">
      <font>
       <family>宋体</family>
      </font>
     </property>
    </column>
   </widget>
   <widget class="QTextEdit" name="text_name">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>60</y>
      <width>261</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>姓名：</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="text_number">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>130</y>
      <width>261</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>140</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>电话号码：</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="text_mail">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>190</y>
      <width>261</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'宋体'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'宋体';&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>邮箱：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>250</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>地址：</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="text_addr">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>240</y>
      <width>651</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_search">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>190</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>查找</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_add">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>60</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>添加</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>120</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>散列函数：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="com_fun">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>110</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>BKDR Hash</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>JS Hash</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>AP Hash</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>RS Hash</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>SDBM Hash</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PJW Hash</string>
     </property>
    </item>
   </widget>
   <widget class="QTextEdit" name="text_file">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>611</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
     </font>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_file">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>10</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>选择文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_rand">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>60</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>生成随机信息</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>400</y>
      <width>761</width>
      <height>231</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
     </font>
    </property>
    <column>
     <property name="text">
      <string>哈希算法</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>冲突解决方法</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>数据量</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>冲突率(%)</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>生成耗时(ms)</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>平均搜索时间(ms)</string>
     </property>
    </column>
   </widget>
   <widget class="QComboBox" name="con_sol">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>150</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>宋体</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>开放定址法</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>链地址法</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>建立公共溢出区</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>160</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>冲突解决方法：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_search_2">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>190</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Agency FB</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="text">
     <string>建立</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="HashSearch.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>btn_file</sender>
   <signal>clicked()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>btn_file_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>578</x>
     <y>26</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_rand</sender>
   <signal>clicked()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>btn_rand_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>543</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_add</sender>
   <signal>clicked()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>btn_add_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>544</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_search</sender>
   <signal>clicked()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>btn_search_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>586</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_search_2</sender>
   <signal>clicked()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>btn_built_click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>221</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>text_name</sender>
   <signal>textChanged()</signal>
   <receiver>HashSearchClass</receiver>
   <slot>text_length()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>btn_file_click()</slot>
  <slot>btn_rand_click()</slot>
  <slot>btn_add_click()</slot>
  <slot>btn_search_click()</slot>
  <slot>btn_built_click()</slot>
  <slot>text_length()</slot>
 </slots>
</ui>
